sdkproject/Assets/Mapbox/Unity/MeshGeneration/Data/UnityTile.cs

```diff
@@ -230,23 +230,23 @@ public Texture2D GetRasterData()
         internal void AddTile(Tile tile)
 		{
 			_tiles.Add(tile);
-            tile.OnTileError += OnTileError;
+            tile.OnTileError += OnTileErrorHandler;
 		}
 
 		public void Cancel()
 		{
 			for (int i = 0, _tilesCount = _tiles.Count; i < _tilesCount; i++)
 			{
 				_tiles[i].Cancel();
-                _tiles[i].OnTileError -= OnTileError;
+                _tiles[i].OnTileError -= OnTileErrorHandler;
 			}
 		}
 
         //OnTileError delegate handle method to bubble up the event all the way up to the chain to AbstractTileFactory and MapVisualizer
-        private void OnTileError(CanonicalTileId id){
-            if (OnTileErrorEvent != null)
+		private void OnTileErrorHandler(TileErrorEventArgs e){
+            if (OnTileError != null)
             {
-                OnTileErrorEvent(id);
+				OnTileError(new TileErrorEventArgs(e.TileId,this,e.Exceptions));
```

Comment:
Because event properties are private set, you cannot simply modify existing event args instance. This leads to additional garbage. I'm wondering if encapsulation for event args offers many benefits?

Reply:
While it doesn't offer much benefits beyond just preventing classes from randomly garbling the data on an existing TileErrorEventArgs instance.
But, I guess that is what we need here anyways. I'll make them public get; set;

Reply:
Another question is, do we need a unitytile instance on the errorevent?

Reply:
Hmm, may be needed for some types of exception handling? Perhaps injecting data into the UnityTile instance after making a new request outside factory framework?